Extension/src/LanguageServer/client.ts

```diff
@@ -948,13 +1038,12 @@ export class DefaultClient implements Client {
 
                     if (isFirstClient) {
                         workspaceReferences = new refs.ReferencesManager(this);
-
                         // The configurations will not be sent to the language server until the default include paths and frameworks have been set.
                         // The event handlers must be set before this happens.
-                        const inputCompilerDefaults: configs.CompilerDefaults = await languageClient.sendRequest(QueryCompilerDefaultsRequest, {});
+                        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
```

Comment:
If `const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);` is always called just before `requestCompiler`, can it be moved within `requestCompiler`?

Reply:
It can't because it is possible that a trusted compiler has been added to the persistent state elsewhere on startup. If so we want that present in our first call to query default compiler